services/offline/src/lib/online-status.ts

```diff
@@ -50,5 +55,21 @@ export function useOnlineStatus(
         }
     }, [updateState])
 
+    useEffect(() => {
+        const lastOnline = localStorage.getItem(lastOnlineKey)
+        // When going online, remove 'lastOnline' if it's set
+        if (online && lastOnline) {
+            localStorage.removeItem(lastOnlineKey)
+        }
+        // When going offline, set 'lastOnline' if it's undefined
+        if (!online && !lastOnline) {
+            localStorage.setItem(
+                lastOnlineKey,
+                // Using Date.now() to simplify testing
+                new Date(Date.now()).toUTCString()
+            )
+        }
+    }, [online])
+
     return { online, offline: !online }
```

Comment:
Interesting note: the way this is implemented currently using `useEffect` to set the `lastOnline` value in localStorage, that suggested code doesn't work to return the most recent `lastOnline` value from the hook because there isn't a rerender once the value in localStorage is changed. I'll have to refactor a bit to couple the localStorage change with the `online` state change so that the hook returns the most up-to-date `lastOnline` value, ideally without another rerender

Reply:
What about using the global store instead of local storage?

Reply:
Using the global store would be a useful part of 'Single online status provider' solution that Austin mentions below, but using just the global store alone wouldn't be able to track the status across sessions